src/clockcache.c

```diff
@@ -1888,15 +1891,12 @@ clockcache_deinit(clockcache *cc) // IN/OUT
 #endif
    }
 
-   if (cc->rc_bh) {
-      platform_buffer_destroy(cc->rc_bh);
-   }
+   platform_buffer_deinit(&cc->rc_bh);
 
    platform_free(cc->heap_id, cc->entry);
```

Comment:
Along similar lines, I think all of these need to be wrapped in NULL tests.

Reply:
That would not be needed if we pushed the NULL-check to inside `platform_buffer_deinit()`. No ?

Reply:
I misunderstood your comment. Yes, you are right; the calls to `platform_free()` will now only be done if the ptr-variable is non-NULL. Fixed.